src/eos-updater-poll-common.c

```diff
@@ -312,6 +310,118 @@ get_booted_refspec (gchar               **booted_refspec,
   return TRUE;
 }
 
+static gboolean
+get_refspec_to_upgrade_on_from_deployment (OstreeSysroot     *sysroot,
+                                           OstreeDeployment  *booted_deployment,
+                                           gchar            **out_refspec_to_upgrade_from_deployment,
+                                           GError           **error)
+{
+  const gchar *checksum = ostree_deployment_get_csum (booted_deployment);
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) metadata = NULL;
+  g_autoptr(GVariant) ref_for_deployment_variant = NULL;
+  g_autofree gchar *refspec_for_deployment = NULL;
+  OstreeRepo *repo = NULL;
+
+  g_return_val_if_fail (out_refspec_to_upgrade_from_deployment != NULL, FALSE);
+
+  if (!ostree_sysroot_get_repo (sysroot, &repo, NULL, error))
+   return FALSE;
+
+  if (!ostree_repo_load_variant (repo,
+                                 OSTREE_OBJECT_TYPE_COMMIT,
+                                 checksum,
+                                 &commit,
+                                 error))
+    return FALSE;
+
+  /* Look up the checkpoint target to see if there is one on this commit. */
+  metadata = g_variant_get_child_value (commit, 0);
+  ref_for_deployment_variant = g_variant_lookup_value (metadata,
+                                                       "eos.checkpoint-target",
+                                                       G_VARIANT_TYPE_STRING);
+
+  /* No metadata tag on this commit, just return TRUE with no value */
+  if (ref_for_deployment_variant == NULL)
+    {
+      *out_refspec_to_upgrade_from_deployment = NULL;
+      return TRUE;
+    }
+
+  refspec_for_deployment = g_variant_dup_string (ref_for_deployment_variant, NULL);
+
+  if (refspec_for_deployment != NULL &&
```

Comment:
I’m pretty sure `refspec_for_deployment` can’t be `NULL` here, since `g_variant_dup_string()` should never return a `NULL` string (it can’t fail, and `GVariant`s can’t contain `NULL` strings).